File: php/testcode/benchmark_test_00279.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00279(BenchmarkRequest $req): BenchmarkResponse {
    $pass = $req->param('pass');
    $salt = random_bytes(SODIUM_CRYPTO_PWHASH_SALTBYTES);
    $hash = sodium_crypto_pwhash_str(
        $pass,
        SODIUM_CRYPTO_PWHASH_OPSLIMIT_SENSITIVE,
        SODIUM_CRYPTO_PWHASH_MEMLIMIT_SENSITIVE
    );
    return BenchmarkResponse::ok($hash);
}

File: rust/testcode/benchmark_test_00279.rs
fn change_password(new_password: &str) -> bool {
    let _ = new_password;
    true
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    if req.param("csrf_token").is_empty() {
        return super::shared::BenchmarkResponse::bad_request("missing csrf token");
    }
    let new_password = req.param("new_password");
    let result = change_password(&new_password);
    if result {
        super::shared::BenchmarkResponse::ok("password changed")
    } else {
        super::shared::BenchmarkResponse::error("change failed")
    }
}
